src/main/java/io/github/syst3ms/skriptparser/expressions/ExprStringCharAt.java

```diff
@@ -28,27 +27,28 @@
 		);
 	}
 
-	private Expression<Integer> position;
+	private Expression<BigInteger> position;
 	private Expression<String> value;
 
 	@SuppressWarnings("unchecked")
 	@Override
-	public boolean init(final Expression<?> @NotNull [] expressions, final int matchedPattern, @NotNull final ParseContext parseContext) {
-		position = (Expression<Integer>) expressions[0];
+	public boolean init(Expression<?>[] expressions, int matchedPattern, ParseContext parseContext) {
+		position = (Expression<BigInteger>) expressions[0];
 		value = (Expression<String>) expressions[1];
 		return true;
 	}
 
 	@Override
-	public String[] getValues(final TriggerContext ctx) {
+	public String[] getValues(TriggerContext ctx) {
 		return DoubleOptional.ofOptional(value.getSingle(ctx), position.getSingle(ctx))
-			.filter((val, pos) -> pos > 0 && pos <= val.length())
-			.mapToOptional((val, pos) -> new String[]{ String.valueOf(val.charAt(pos - 1)) } )
+			.filter((val, pos) -> pos.compareTo(BigInteger.ZERO) > 0
```

Comment:
For checking a BigInteger's relation to zero, use its signum() method instead, it's somewhat cheaper.

Reply:
Then we'll have to change a lot of occurrences on this. I used this most of the time I think in the Number API update.

Either way, I'll add it here and we can change that later if deemed necessary.